src/content/docs/query-your-data/nrql-new-relic-query-language/nrql-query-tutorials/NRQL-high-cardinality-metrics.mdx

```diff
@@ -0,0 +1,123 @@
+
+## Examples and sample workflows
+
+If you hit one of your cardinality limits, there are a couple of options you can use to remedy the situation. One easy answer is increasing your limits, but if you would prefer not to do that, a good alternative is to explore which dimensions are contributing the most to your cardinality and think about removing them if they do not provide you value. This can save storage and bandwidth costs and potentially prevent you needing to raise your limits.  
+
+### Find cardinality contributors: metrics
+
+Above, you saw how to query for the current cardinality of a particular metric using `cardinality(memory.heap)`. Here's what that looks like again:
+
+```
+FROM Metric SELECT cardinality(memory.heap) SINCE today RAW
+```
+
+For the total account cardinality, you can use the same basic query structure and simply omit the metric name:
+
+```
+FROM Metric SELECT cardinality() SINCE today RAW
+```
+
+Finally, the account cardinality is essentially the sum of each metric's cardinality, so adding in a simple FACET query can help find the highest cardinality metric:
+
+```
+FROM Metric SELECT cardinality() SINCE today RAW FACET metricName
+```
+### Finding cardinality contributors: dimensions
+
+The next step is to determine which dimensions in a given metric contribute the most to its cardinality. You can do this by checking your current dimensions then removing dimensions to check their impact on cardinality. 
+
+First, let's look at ways you can inspect existing dimensions. This query shows the attributes map of each metric that was sent:
+
+```
+FROM Metric SELECT dimensions() WHERE metricName = 'memory.heap' SINCE today RAW
+```
+
+Looking through these could reveal some dimensions containing a unique ID or other highly variable  value that might be worth removing. 
+
+Next, you can look at `keySet()`:
+
+```
+FROM Metric SELECT keySet() WHERE metricName = 'memory.heap' SINCE today RAW
+```
+
+Understanding the dimensions that have the most influence on your cardinality comes down to understanding how each key's values correlate with each other. You can experiment with what your cardinality would be without a dimension simply by adding it to the exclude list:
+
+```
+FROM Metric SELECT cardinality(memory.heap, exclude: {'container.id'}) SINCE today RAW
+```
+
+Likewise, there is an include list if that is more convenient to the query context:
+
+```
+FROM Metric SELECT cardinality(memory.heap, include: {'host.name', 'region'}) SINCE today RAW
+```
+
+To summarize, the more familiar you are with your data, the easier it will be to know which keys to include or exclude comes down to familiarity with your data. Things that look like unique identifiers are a good place to start, but sometimes it's no single key but the combination of two or more keys. 
```

Comment:
Perhaps this deserves its own Conclusion header?

```suggestion
Managing cardinality can be tricky to conceptualize, but the above tools should help you get answers to questions like "What metric is contributing the most cardinality" and "What impact does a given attribute(s) have to that total cardinality". It is often the case that cardinality tracks with the most unique value, as that value may pin down the possible values other attributes can take on. However, there are plenty of cases where the explosion of possible combinations of a handful of attributes drives the total cardinality. The more familiar you are with your data and the systems that generate it, the easier it will be to know which attributes to include or exclude. Things that look like unique identifiers are generally a good place to start, but sometimes it's no single key but the combination of two or more keys.
```